llama_stack/providers/utils/inference/model_registry.py

```diff
@@ -79,28 +79,28 @@ async def register_model(self, model: Model) -> Model:
             provider_resource_id = model.provider_resource_id
         else:
             provider_resource_id = self.get_provider_model_id(model.provider_resource_id)
+
         if provider_resource_id:
             model.provider_resource_id = provider_resource_id
         else:
-            if model.metadata.get("llama_model") is None:
-                raise ValueError(
-                    f"Model '{model.provider_resource_id}' is not available and no llama_model was specified in metadata. "
-                    "Please specify a llama_model in metadata or use a supported model identifier"
-                )
+            llama_model = model.metadata.get("llama_model")
+            if llama_model is None:
+                return model
+
```

Comment:
I think this change also allows non-Llama models for remote vLLM provider as well, right?

Reply:
@terrytangyuan indeed it does. this is the key change.